examples/ConversionForm/CollectionInput.tsx

```diff
@@ -0,0 +1,153 @@
+    dataType: string;
+}
+
+interface CollectionState {
+    length: number;
+}
+
+class CollectionInput extends React.Component<
+    CollectionInputProps,
+    CollectionState
+> {
+    constructor(props: CollectionInputProps) {
+        super(props);
+        this.state = {
+            length: props.parameter.length,
+        };
+        this.handleChange = this.handleChange.bind(this);
+        this.addItem = this.addItem.bind(this);
+    }
+
+    handleChange(key, index, type, targetValue) {
+        const { handler, id } = this.props;
+        const newState = { ...this.state };
+        if (!newState[index]) {
+            newState[index] = {
+                key: "",
+                value: {},
+            };
+        }
+        if (type === "key") {
+            newState[index] = {
+                ...newState[index],
+                [type]: targetValue,
+            };
+        } else {
+            newState[index] = {
+                ...newState[index],
+                [type]: {
+                    ...newState[index][type],
+                    [key]: targetValue,
+                },
+            };
+        }
+        this.setState(newState);
+        const newValues = reduce(
+            newState,
+            (acc, cur: { key: any; value: any }) => {
+                const key = cur.key;
+                const value = cur.value;
+                acc[key] = value;
+                return acc;
+            },
+            {}
+        );
+        handler([id], index, newValues);
+    }
+
+    renderValueItem = (childItem, newPath, index, type) => {
+        const { templateData } = this.props;
+        const currentDataType = childItem.data_type;
+        const data = templateData[childItem.data_type];
+
+        if (data.isBaseType) {
+            return (
+                <BaseInput
+                    options={childItem.options || []}
+                    dataType={currentDataType}
+                    name={childItem.name}
+                    handler={(value) =>
+                        this.handleChange(
+                            newPath[newPath.length - 1],
+                            index,
+                            type,
+                            value
+                        )
+                    }
+                />
+            );
+        } else if (data.parameters) {
+            return map(data.parameters, (childParameter, key) => {
+                return this.renderValueItem(
+                    childParameter,
+                    [...newPath, key],
+                    index,
+                    type
+                );
+            });
+        }
+    };
+
+    addItem() {
+        const index = this.state.length + 1;
+        this.setState({ length: index });
+    }
+
+    render() {
+        const { name, path, parameter } = this.props;
+        const jsx = [] as JSX.Element[];
+        for (let index = 0; index < this.state.length; index++) {
+            jsx.push(
+                <div>
```

Comment:
should this element take a unique `key` property to go in an array? as in `<div key={index}>`